File: backend/src/main/java/org/example/backend/monthPlan.java
package org.example.backend;

import java.time.YearMonth;
import java.util.List;
import java.util.Map;


public record monthPlan(
        String user,
        YearMonth yearMonth,
        double totalBudget,
        double totalLeftover,
        Map<String, CategoryPlan> categoryPlanMap,
        List<Transaction> transactions
) {
}

File: frontend/src/types/MonthPlan.ts
export type MonthPlan = {
    id:string,
    user:string
    yearMonth:string,
    totalBudget:number,
    totalLeftover:number,
    categoryPlans: CategoryPlan[],
    transactions: Transaction[]
}

export type CategoryPlan = {
    categoryBudget:string,
    categoryLeftover:number
}

export type Transaction={
    id:string,
    timestamp:string,
    user:string,
    amount:number,
    note:string,
    category:string
}